base/src/notification-manager/notification-manager.ts

```diff
@@ -9,33 +9,55 @@
   Copyright Contributors to the Zowe Project.
 */
 
+import { ZoweNotification } from './notification';
 
-export class NotificationManager implements MVDHosting.NotificationManagerInterface {
+export class ZoweNotificationManager implements MVDHosting.ZoweNotificationManagerInterface {
   private notificationCache: any[];
-  private handlers: MVDHosting.NotificationWatcher[];
+  private handlers: MVDHosting.ZoweNotificationWatcher[];
+  private ws: WebSocket;
+  private url: string;
 
   constructor() {
-    this.notificationCache = new Array<Notification>();
-    this.handlers = new Array<MVDHosting.NotificationWatcher>();
+    this.notificationCache = new Array<ZoweNotification>();
+    this.handlers = new Array<MVDHosting.ZoweNotificationWatcher>();
   }
 
-  push(notification: Notification): void {
-    this.notificationCache.push(notification);
-    for (let i = 0; i < this.handlers.length; i++) {
-      this.handlers[i].handleMessageAdded();
+  _setURL(url: string): void {
+    this.url = url;
+    this.ws = new WebSocket(url);
+    var _this = this;
+    this.ws.onmessage = function(message) {
+      _this.notificationCache.push((JSON.parse(message.data)['notification']) as ZoweNotification)
+      _this.updateHandlers(JSON.parse(message.data));
+    }
+    this.ws.onclose = function() {
+      _this.ws = new WebSocket(this.url)
+    }
+    this.ws.onerror = function() {
+      _this.ws = new WebSocket(this.url)
     }
   }
 
-  pop(): Notification | void {
-    let n = this.notificationCache.pop();
+  updateHandlers(message: any): void {
     for (let i = 0; i < this.handlers.length; i++) {
-      this.handlers[i].handleMessageAdded();
+      this.handlers[i].handleMessageAdded(message, this.notificationCache.length -1);
     }
-    return n;
   }
 
-  getAll(): Notification[] {
-    let copy: Notification[] = this.notificationCache.slice(0);
+  getURL() {
+    return this.url;
```

Comment:
Do you actually need this? If you do, then disregard my above comment. But I think you dont.